File: jobs/2026-05-30__07-57-35/cron_job_cleanup_go__4gPkNog/artifacts/user/cleanup-app/cleanup/cleanup.go
package cleanup

import (
	"context"
	"time"

	"encore.dev/storage/sqldb"
)

var db = sqldb.NewDatabase("cleanup_db", sqldb.DatabaseConfig{
	Migrations: "./migrations",
})

type Record struct {
	ID        string    `json:"id"`
	Data      string    `json:"data"`
	CreatedAt time.Time `json:"created_at"`
}

type InsertRecordRequest struct {
	ID        string    `json:"id"`
	Data      string    `json:"data"`
	CreatedAt time.Time `json:"created_at"`
}

//encore:api public method=POST path=/records
func InsertRecord(ctx context.Context, req *InsertRecordRequest) error {
	_, err := db.Exec(ctx, `
		INSERT INTO records (id, data, created_at)
		VALUES ($1, $2, $3)
	`, req.ID, req.Data, req.CreatedAt)
	return err
}

type ListRecordsResponse struct {
	Records []Record `json:"records"`
}

//encore:api public method=GET path=/records
func ListRecords(ctx context.Context) (*ListRecordsResponse, error) {
	rows, err := db.Query(ctx, `SELECT id, data, created_at FROM records`)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	var records []Record
	for rows.Next() {
		var r Record
		if err := rows.Scan(&r.ID, &r.Data, &r.CreatedAt); err != nil {
			return nil, err
		}
		records = append(records, r)
	}

	if err := rows.Err(); err != nil {
		return nil, err
	}

	if records == nil {
		records = []Record{}
	}

	return &ListRecordsResponse{Records: records}, nil
}

//encore:api public method=POST path=/cleanup
func CleanupStaleRecords(ctx context.Context) error {
	cutoff := time.Now().Add(-24 * time.Hour)
	_, err := db.Exec(ctx, `DELETE FROM records WHERE created_at < $1`, cutoff)
	return err
}

File: jobs/2026-05-30__07-57-35/cron_job_cleanup_ts__2Qhk9Hc/artifacts/user/cron-app-zr-2qhk9hc/cleanup/cleanup.ts
import { api } from "encore.dev/api";
import { SQLDatabase } from "encore.dev/storage/sqldb";
import { CronJob } from "encore.dev/cron";
import type { IncomingMessage, ServerResponse } from "node:http";

const db = new SQLDatabase("records_db", {
  migrations: "./migrations",
});

interface Record {
  id: number;
  data: string;
  created_at: string;
}

interface InsertRecordRequest {
  data: string;
  created_at?: string;
}

interface InsertRecordResponse {
  id: number;
}

export const insertRecord = api(
  { expose: true, method: "POST", path: "/records" },
  async (req: InsertRecordRequest): Promise<InsertRecordResponse> => {
    let row;
    if (req.created_at) {
      row = await db.queryRow`
        INSERT INTO records (data, created_at)
        VALUES (${req.data}, ${req.created_at}::timestamptz)
        RETURNING id
      `;
    } else {
      row = await db.queryRow`
        INSERT INTO records (data)
        VALUES (${req.data})
        RETURNING id
      `;
    }
    return { id: row!.id };
  }
);

export const getRecords = api.raw(
  { expose: true, method: "GET", path: "/records" },
  async (req: IncomingMessage, res: ServerResponse) => {
    try {
      const rows = await db.query`
        SELECT id, data, created_at
        FROM records
        ORDER BY id
      `;
      const records: Record[] = [];
      for await (const row of rows) {
        records.push({
          id: row.id,
          data: row.data,
          created_at: row.created_at,
        });
      }
      res.setHeader("Content-Type", "application/json");
      res.end(JSON.stringify(records));
    } catch (err) {
      res.statusCode = 500;
      res.end(JSON.stringify({ error: String(err) }));
    }
  }
);

interface CleanupResponse {
  deleted_count: number;
}

export const cleanupRecords = api(
  { expose: true, method: "POST", path: "/cleanup" },
  async (): Promise<CleanupResponse> => {
    // Delete records older than 1 hour
    const row = await db.queryRow`
      WITH deleted AS (
        DELETE FROM records
        WHERE created_at < NOW() - INTERVAL '1 hour'
        RETURNING id
      )
      SELECT COUNT(*) as count FROM deleted
    `;
    return { deleted_count: Number(row?.count || 0) };
  }
);

const cleanupJob = new CronJob("cleanup-job", {
  title: "Cleanup Job",
  every: "1h",
  endpoint: cleanupRecords,
});
